packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -377,16 +394,42 @@ export function sendTransaction<
 	ResolveType = FormatType<ReceiptInfo, ReturnFormat>,
 >(
 	web3Context: Web3Context<EthExecutionAPI>,
-	transaction: Transaction,
+	transaction: Transaction | TransactionWithLocalWalletIndex,
 	returnFormat: ReturnFormat,
 	options?: SendTransactionOptions<ResolveType>,
 ): PromiEvent<ResolveType, SendTransactionEvents> {
-	let transactionFormatted = formatTransaction(transaction, DEFAULT_RETURN_FORMAT);
-
 	const promiEvent = new PromiEvent<ResolveType, SendTransactionEvents>((resolve, reject) => {
 		setImmediate(() => {
 			(async () => {
 				try {
+					let transactionFrom: string | undefined;
+					if (transaction.from) {
+						if (typeof transaction.from === 'string' && isAddress(transaction.from)) {
+							transactionFrom = transaction.from;
+						} else if (isNumber(transaction.from)) {
+							if (web3Context.wallet) {
+								transactionFrom = web3Context.wallet.get(
+									format({ eth: 'uint' }, transaction.from, {
+										number: FMT_NUMBER.NUMBER,
+										bytes: FMT_BYTES.HEX,
+									}),
+								).address;
+							} else {
+								throw new LocalWalletNotAvailableError();
+							}
+						} else {
+							throw new InvalidTransactionWithSender(transaction.from);
+						}
+					}
+
+					let transactionFormatted = formatTransaction(
+						{
+							...transaction,
+							from: transactionFrom,
+						},
+						DEFAULT_RETURN_FORMAT,
+					);
+
```

Comment:
`formatTransaction` doesn't support `transaction.from` being anything other than `HexString | undefined`, and it doesn't make sense for it to be updated to handle a local wallet index, because then it would need access to `Web3Context` or a wallet. So instead, this code is enforcing that `transactionFrom` is either a `string | undefined` where the `string` is sourced from either `transaction.from` (if it's a `HexString` address, we could consider allowing this to be other valid bytes values) or `web3Context.wallet.get().address` - this code was a result of writing a `sendTransaction` test with a index for `transaction.from`

Reply:
FYI this is was abstracted a little bit to make use of the [getTransactionFromAttr](https://github.com/ChainSafe/web3.js/blob/d7e88f645e086ebb82764a8d31835fd7f149fd48/packages/web3-eth/src/utils/transaction_builder.ts#L50) method